mcs/class/System.Private.CoreLib/System.Runtime.InteropServices/NativeLibrary.cs

```diff
@@ -1,12 +1,24 @@
+using System.IO;
 using System.Reflection;
 
 namespace System.Runtime.InteropServices
 {
 	partial class NativeLibrary
 	{
+		[DllImport ("libdl")]
+		private static extern IntPtr dlopen (string libName, int flags);
+		
+		private const int RTLD_LAZY = 0x001;
+		
 		static IntPtr LoadLibraryByName (string libraryName, Assembly assembly, DllImportSearchPath? searchPath, bool throwOnError) => throw new NotImplementedException ();
 
-		static IntPtr LoadFromPath (string libraryName, bool throwOnError) => throw new NotImplementedException ();
+		static IntPtr LoadFromPath (string libraryName, bool throwOnError) {
```

Comment:
Had to implement it because `PlatformDetection` in corefx now uses it to load some gdi library: https://github.com/dotnet/corefx/blob/master/src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs#L66-L78

Reply:
Implementing it this way will break the Windows builds. :-/

Reply:
oh I always forget 😄copy-pasted the Unix impl for tests

Reply:
I am fine if it remains unimplemented (ie. always returns error) but I would prefer not to break the builds ;)